tests/meta/archive/test_archive_api.py

```diff
@@ -167,3 +166,59 @@ def test_call_corresponding_archiving_method(spring_archive_api: SpringArchiveAP
 
     # THEN the correct archive function should have been called once
     mock_request_submitter.assert_called_once_with(files_and_samples=[file_and_sample])
+
+
+def test_archive_all_non_archived_spring_files(
+    spring_archive_api: SpringArchiveAPI, caplog, ok_ddn_response
+):
+    """Test archiving all non-archived SPRING files for Miria customers."""
+    # GIVEN a populated status_db database with two customers, one DDN and one non-DDN,
+    # with the DDN customer having two samples, and the non-DDN having one sample.
+
+    # WHEN archiving all available files
+    with mock.patch.object(
+        AuthToken,
+        "model_validate_json",
+        return_value=AuthToken(
+            access="test_auth_token",
+            expire=int((datetime.now() + timedelta(minutes=20)).timestamp()),
+            refresh="test_refresh_token",
+        ),
+    ), mock.patch.object(
+        APIRequest,
+        "api_request_from_content",
+        return_value=ok_ddn_response,
+    ) as mock_request_submitter:
+        spring_archive_api.archive_all_non_archived_spring_files()
+
+    # THEN the DDN archiving function should have been called with the correct destination and source.
+    mock_request_submitter.assert_called_with(
```

Comment:
This probably belongs in a fixture, no need to have all this setup here

Reply:
@seallard Is there a way of putting the input for a method in a fixture?

Reply:
Aha! It just looked like a lot of data polluting the test. If you feel like it and think it is worth the effort maybe some fixtures could be extracted for the different args.

This one for instance

```json
{
            "osType": "Unix/MacOS",
            "createFolder": False,
            "pathInfo": [
                {
                    "destination": "archive@repository:ADM1",
                    "source": "local@storage:/tests/fixtures/apps/demultiplexing/demultiplexed-runs/spring/dummy_run_001.spring",
                }
            ],
            "metadataList": [],
        }
```

Reply:
Will do that!